packages/data-stores/src/queries/use-jetpack-search-ai.ts

```diff
@@ -24,6 +24,7 @@ export type JetpackSearchAIResult = {
 	step: string;
 	urls: AIResponseURL[];
 	terms: string[];
+	source: null | string;
```

Comment:
To avoid duplicating the `data?.source | 'sitebot'` checks below, you could add [a `select` callback to the `useQuery` call](https://tanstack.com/query/v4/docs/react/reference/useQuery) below and make sure the `source` field is set correctly. Then you could even adjust the type here to be always `string`.

Example from a different part of Calypso code:
https://github.com/Automattic/wp-calypso/blob/61d721e66a191447821a02c29684cf65772d67ee/packages/data-stores/src/support-queries/use-support-activity.ts#L26

Props to @alshakero for pointing out this feature of `useQuery`.

Something like this:
```
select: ( data ) => {
	return { ...data, source: data.source ?? 'sitebot', };
},
```

Reply:
TIL about select. Awesome, this change really simplifies the code!